### PR TITLE
[JENKINS-42556] Improved logging for Queue

### PR DESCRIPTION
**Pull request by jglick**
Logging improvements useful in the course of developing a fix for [JENKINS-42556](https://issues.jenkins-ci.org/browse/JENKINS-42556).

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/model/Queue.java [hunk 1]
```diff
@@ -728,7 +728,11 @@ public boolean cancel(Task p) {
     }
 
     private void updateSnapshot() {
-        snapshot = new Snapshot(waitingList, blockedProjects, buildables, pendings);
+        Snapshot revised = new Snapshot(waitingList, blockedProjects, buildables, pendings);
+        if (LOGGER.isLoggable(Level.FINEST)) {
+            LOGGER.log(Level.FINEST, "{0} → {1}; leftItems={2}", new Object[] {snapshot, revised, leftItems.asMap()});
```
Comment (jglick) [inline core/src/main/java/hudson/model/Queue.java]: Generally handy to know when the state of the queue changes, for whatever reason.

File: core/src/main/java/hudson/model/Queue.java [hunk 2]
```diff
@@ -1443,9 +1447,11 @@ public void maintain() {
                 }
                 // pending -> buildable
                 for (BuildableItem p: lostPendings) {
-                    LOGGER.log(Level.FINE,
+                    if (LOGGER.isLoggable(Level.FINE)) {
```
Comment (jglick) [inline core/src/main/java/hudson/model/Queue.java]: Avoid computing `getFullDisplayName` if we do not have to.

File: core/src/main/java/hudson/model/Queue.java [hunk 3]
```diff
@@ -1464,7 +1470,7 @@ public void maintain() {
                     Collections.sort(blockedItems, QueueSorter.DEFAULT_BLOCKED_ITEM_COMPARATOR);
                 }
                 for (BlockedItem p : blockedItems) {
-                    String taskDisplayName = p.task.getFullDisplayName();
+                    String taskDisplayName = LOGGER.isLoggable(Level.FINEST) ? p.task.getFullDisplayName() : null;
```
Comment (jglick) [inline core/src/main/java/hudson/model/Queue.java]: I recommended @christ66 do this in #1863 but he did not. Turns out this had a far worse consequence than wasting a bit of CPU time, which will become clearer in another PR.

File: core/src/main/java/hudson/model/Queue.java (hunk 2 above)
Comment (jglick) [inline core/src/main/java/hudson/model/Queue.java]: Note that there are more cases of this optimization that could be done; I just picked some sample cases.

**Review by aheritier — APPROVED** [review]

**Comment by aheritier** [comment]
🐝

**Review by oleg-nenashev — APPROVED** [review]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done. I consider merging it since any performance optimization of the queue is useful

File: core/src/main/java/hudson/model/Queue.java (hunk 2 above)
Comment (jglick) [inline core/src/main/java/hudson/model/Queue.java]: #2947

**Pull request merged**